### ISSUE TITLE
Faker issue after python update to 3.8.10

### ISSUE THREAD

**Issue by viktor-zinchenko** [joke2k/faker#1438]
* Faker version: 6.6.3
* OS: MacOs 10.15.4
* Python version: 3.8.10

After python from update from 3.8.9 to 3.8.10 got and issue with faker.

### Steps to reproduce

1. Install Faker and pytest as a project dependencies.
2. Run: `pytest`

Traceback:

```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/pkgutil.py", line 415, in get_importer
    importer = sys.path_importer_cache[path_item]
KeyError: PosixPath('/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/faker/providers')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/user/Documents/work/work/venv/project/bin/pytest", line 8, in <module>
    sys.exit(console_main())
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/config/__init__.py", line 185, in console_main
    code = main()
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/config/__init__.py", line 143, in main
    config = _prepareconfig(args, plugins)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/config/__init__.py", line 318, in _prepareconfig
    config = pluginmanager.hook.pytest_cmdline_parse(
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/helpconfig.py", line 100, in pytest_cmdline_parse
    config: Config = outcome.get_result()
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/config/__init__.py", line 1003, in pytest_cmdline_parse
    self.parse(args)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/config/__init__.py", line 1283, in parse
    self._preparse(args, addopts=addopts)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/config/__init__.py", line 1172, in _preparse
    self.pluginmanager.load_setuptools_entrypoints("pytest11")
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/pluggy/manager.py", line 299, in load_setuptools_entrypoints
    plugin = ep.load()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/assertion/rewrite.py", line 170, in exec_module
    exec(co, module.__dict__)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/faker/__init__.py", line 1, in <module>
    from faker.factory import Factory
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/assertion/rewrite.py", line 170, in exec_module
    exec(co, module.__dict__)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/faker/factory.py", line 7, in <module>
    from faker.config import AVAILABLE_LOCALES, DEFAULT_LOCALE, PROVIDERS
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/assertion/rewrite.py", line 170, in exec_module
    exec(co, module.__dict__)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/faker/config.py", line 11, in <module>
    PROVIDERS = find_available_providers(
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/faker/utils/loading.py", line 57, in find_available_providers
    for mod in list_module(providers_mod) if mod != '__pycache__'
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/faker/utils/loading.py", line 35, in list_module
    return [name for _, name, is_pkg in pkgutil.iter_modules([path]) if is_pkg]
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/faker/utils/loading.py", line 35, in <listcomp>
    return [name for _, name, is_pkg in pkgutil.iter_modules([path]) if is_pkg]
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/pkgutil.py", line 129, in iter_modules
    for i in importers:
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/pkgutil.py", line 419, in get_importer
    importer = path_hook(path_item)
  File "<frozen importlib._bootstrap_external>", line 1594, in path_hook_for_FileFinder
  File "<frozen importlib._bootstrap_external>", line 1469, in __init__
  File "<frozen importlib._bootstrap_external>", line 177, in _path_isabs
AttributeError: 'PosixPath' object has no attribute 'startswith'
make: *** [pytest] Error 1
```

### Expected behavior

pytest should work

### Actual behavior

`AttributeError: 'PosixPath' object has no attribute 'startswith'`

p.s. Could be caused by [python update to 3.8.10](https://github.com/python/cpython/commit/eed7686e9fe22a3eb5e1a1fc7d27c27fca070bd1)

**Comment by jaspalpuri-zopa** [comment]
I had the same issue this morning. Upgrading to faker==8.1.2 resolved it for me, however.

**Comment by The-Compiler** [comment]
According to the [changelog](https://github.com/joke2k/faker/blob/master/CHANGELOG.md#v701---2021-03-29) this was fixed in faker v7.0.1 - so this seems like a duplicate of #1421 and fixed by #1423.

**Comment by fcurella** [comment]
Closing as duplicate
